templates/terraform/examples/apigateway_api_basic.tf.erb

```diff
@@ -0,0 +1,8 @@
+resource "google_api_gateway_api" "<%= ctx[:primary_resource_id] %>" {
```

Comment:
Highly recommend breaking this up into two tests: one "basic" which only sets required fields (this is important so we can tests that *not* setting certain fields works), and another "full" which sets as many fields as you can. Then you also should add a third handwritten update test that updates every single field that can be updated (my default is to go from unset to set and then back so I can reuse the configs from the basic/full tests, but some APIs lend themselves to different approaches)